clients/src/test/java/org/apache/kafka/common/metrics/JmxReporterTest.java

```diff
@@ -48,7 +50,10 @@ public void testJmxRegistration() throws Exception {
             assertTrue(server.isRegistered(new ObjectName(":type=grp2")));
             assertEquals(0.0, server.getAttribute(new ObjectName(":type=grp2"), "pack.bean2.total"));
 
-            metrics.removeMetric(metrics.metricName("pack.bean1.avg", "grp1"));
+            KafkaMetric metric = metrics.removeMetric(metrics.metricName("pack.bean1.avg", "grp1"));
+            MetricName metricName = metric.metricName();
+            String mBeanName = JmxReporter.getMBeanName("", metricName);
+            assertFalse(reporter.containsMBean(mBeanName));
```

Comment:
Let's also verify that `containsMBean` returns  `true` before the removal. Also, it wouldn't hurt to check that the second `removeMetric` call removes it from `JmxReporter`.